pkg/nerdgraph/typegen.go

```diff
@@ -7,6 +7,10 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+var (
+	types = make(map[string]string)
```

Comment:
Does this need to be package global?

Reply:
I'm trying to figure that out now.

Reply:
Currently, yes.  The recursion makes it such that the global is updated when a new type is seen, so we don't end up with duplicates, or multiple fetches for the same time.